src/main/java/gregtech/api/multiblock/BlockPatternChecker.java

```diff
@@ -0,0 +1,49 @@
+package gregtech.api.multiblock;
+
+import codechicken.lib.vec.Vector3;
+import gregtech.api.metatileentity.multiblock.MultiblockControllerBase;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+
+public class BlockPatternChecker {
+    
+    public static BlockPos getPatternErrorPos(MultiblockControllerBase controllerBase) {
+        BlockPattern structurePattern = controllerBase.getStructurePattern();
+        if(structurePattern != null && structurePattern.checkPatternAt(controllerBase.getWorld(), controllerBase.getPos(), controllerBase.getFrontFacing().getOpposite()) == null) {
+            return new BlockPos(structurePattern.getBlockPos());
```

Comment:
Why didn't you make BlockPattern do this convert to immutable?

It seems like a bad idea to expose that mutable state from BlockPattern for others to corrupt.

The same goes for structureDir (the elements will be mutable).

Reply:
I copied most of the code from Gregicality, where I created a new object to record it so it didn't have this problem there. However, you are right, I simplified the implementation, which is a real potential bug. The reason I wrote it outside the BlockPattern is that I want to keep the function name available in Gregicality. ill convert it to draft and make it more elegant